### PR TITLE
feat: Merge data and row height menus in explist v2

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This merges the data and row height menus together into one menu. The icon used for the infinite scroll menu item is brought into the icon component. The icon component is also refactored a little to ensure that all svg icons are accounted for and won't overlap with the font icons.

## Test Plan

with the experiment v2 feature flag on and visiting the experiment list page:
* there should be a single miscellaneous options menu on the table
* selecting a row height menu option should select the menu option and close the menu
* clicking the infinite scroll menu option should switch the data scroll method and close the menu

## Checklist

- [X] Changes have been manually QA'd
- [X] User-facing API changes need the "User-facing API Change" label.
- [X] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [X] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1426

### CODE REVIEW

File: webui/react/src/components/kit/Icon.tsx [hunk 1]
```diff
@@ -202,7 +214,32 @@ const FourSquaresIcon: React.FC = () => (
...
+      <line strokeOpacity="0.72" x1="4.5" x2="12.7" y1="9.5" y2="9.5" />
+      <line strokeOpacity="0.48" x1="4.5" x2="12.7" y1="13.5" y2="13.5" />
+      <line strokeOpacity="0.24" x1="4.5" x2="12.7" y1="17.5" y2="17.5" />
+      <path d="M20 15.7059L17.6 18M17.6 18L15.2 15.7059M17.6 18V5" />
+    </g>
+  </svg>
+);
+
+// intersection here is to ensure the index access in the component returns
+// undefined | React.FC and not any
```
Comment (gt2345) [inline webui/react/src/components/kit/Icon.tsx:230]: :clap:

**Review by gt2345 — APPROVED** [review]
LGTM

**Comment by ashtonG** [comment]
@alexjdetermined could you take a quick look at this? as asked, the menus are merged, but there are slight differences in the infinite scroll menu item:
- the option now changes when clicking directly on the menu item instead of only on the toggle
- the icon is now responsive to dark/light mode

image
image

**Review by alexjdetermined — APPROVED** [review]
LGTM

**Pull request merged**